### PR TITLE
Design Doc for Nvidia GPU Request Intergration

### PR DESCRIPTION
**Pull request by 5had3z**
Separate PR for design doc in https://github.com/volcano-sh/volcano/pull/1934

### CODE REVIEW

**Comment by Thor-wl** [comment]
Thanks for your contribution! Once the CI has fixed, I'll approve and run the CI.

File: docs/design/nvidia-gpu-operator.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Exclusive GPU sharing currently doesn't take advantage of the work already done by Nvidia's GPU operator in finding GPUs and adding node annotations. Using these annotations foregoes the effort of running our own device plugins for device discovery. This also has the advantage of forcing users to only be exposed to the scheduled of GPUs in their container rather than asking them to use the environment variables that are annotated by previous implementations.
+
+## Goals
+ - Support for exclusive access scheduling with based on the standard annotations provided by Nvidia.
+ - Limited user effort other than following Nvidia instructions for creating gpu cluster and changing volcano-scheduler-configmap.
+ - Add this feature with minimised required changes to existing codebase.
+
+## Non-Goals
+ - Support for gpu-sharing and mixed node annotation support (nodes should have the same type of GPU on them)
+
```
Comment (k82cn) [inline docs/design/nvidia-gpu-operator.md:16]: hm... what are we going to do for this proposal?

File: docs/design/nvidia-gpu-operator.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+## Usage
+Follow [how_to_use_gpu_requests](../user-guide/how_to_use_gpu_requests.md) to install gpu-operator and make the relevant changes to configmaps.
```
Comment (hwdef) [inline docs/design/nvidia-gpu-operator.md:19]: 404

Comment (5had3z) [reply docs/design/nvidia-gpu-operator.md:19]: This is in the [full pr](https://github.com/5had3z/volcano/blob/feature/gpu-numbers/docs/user-guide/how_to_use_gpu_requests.md) this pr just has the design doc.

File: docs/design/nvidia-gpu-operator.md (hunk 1 above)
Comment (5had3z) [reply docs/design/nvidia-gpu-operator.md:16]: Using requests should always gives exclusive access to that resource. Without MIG there's no real way to properly allocate fractions of a card exclusively. The current "gpu sharing" doesn't actually enforce gpu memory sharing limits, any user can just request 1MB but use the whole card, nor does this take into consideration compute util vs memory util.

Requesting a specific card on a mixed card node would be possible but would not integrate with gpu operator  resource request handling. Would have to go back to writing custom annotators and allocators.

**Comment by Thor-wl** [comment]
Pls help take a review. Thanks! If possible, I plan to put this design into this weekly meeting for a statement. @william-wang @shinytang6 @k82cn

**Comment by 5had3z** [comment]
> Pls help take a review. Thanks! If possible, I plan to put this design into this weekly meeting for a statement. @william-wang @shinytang6 @k82cn

If this is during outside of work hours AEST-wise I wouldn't mind discussing as well and making changes as needed.

**Comment by 5had3z** [comment]
Just a brief update, nvidia now have an [oversubscription model](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/gpu-sharing.html) in their gpu-operator that could be intergrated into this.

**Comment by l1n** [comment]
This should not be marked stale, can someone from Volcano team please review? This would be incredibly useful.

**Pull request closed without merging**